yarn-project/pxe/src/note_processor/note_processor.ts

```diff
@@ -115,7 +115,9 @@ export class NoteProcessor {
       const { txLogs } = encryptedL2BlockLogs[blockIndex];
       const blockContext = l2BlockContexts[blockIndex];
       const block = blockContext.block;
-      const dataEndIndexForBlock = block.header.state.partial.noteHashTree.nextAvailableLeafIndex;
+      const dataStartIndexForBlock =
+        block.header.state.partial.noteHashTree.nextAvailableLeafIndex -
+        block.body.numberOfTxsIncludingPadded * MAX_NEW_NOTE_HASHES_PER_TX;
```

Comment:
Had to do this change because `txLogs.length` now does not contain logs of padding txs and the for this reason the original formula stopped working. Anyway I think it's more understandable now.